DuckDuckGo/TabViewController.swift

```diff
@@ -245,7 +247,55 @@ class TabViewController: UIViewController {
             jsAlertController.dismiss(animated: false)
         }
     }
-    
+
+    private var navigationExpectationTimer: Timer? {
+        willSet {
+            navigationExpectationTimer?.invalidate()
+        }
+    }
+
+    private static let navigationExpectationInterval = 3.0
+    private func scheduleNavigationExpectation(destinationURL: URL?, onSessionRestored: (() -> Void)? = nil) {
+        // continuous schedule calls may mean user is trying to hit all the buttons while we‘re waiting
+        guard !forceExpectedNavigationIfNeeded() else { return }
```

Comment:
I get a lot of expirations triggered from here when doing regular navigations - I don't think that is expected.

Reply:
Do you have a particular example for this or just random occurrences?

Reply:
Just navigating using URL bar. I'm running this on iOS 15.5 simulator.

Reply:
these expirations must‘ve been caused by recursive requests sent from `webView:decidePolicyForNavigationAction:`, should be fixed now by resetting the timer in this method